src/sched.rs

```diff
@@ -225,6 +225,221 @@ mod sched_linux_like {
+            SCHED_IDLE,
+            #[cfg(target_os = "android")]
+            SCHED_DEADLINE,
+            #[cfg(target_os = "illumos")]
+            SCHED_SYS,
+            #[cfg(target_os = "illumos")]
+            SCHED_IA,
+            #[cfg(target_os = "illumos")]
+            SCHED_FSS,
+            #[cfg(target_os = "illumos")]
+            SCHED_FX,
+        }
+
+        impl TryFrom<i32>
+    }
+
+    libc_bitflags! {
+        pub struct SchedFlags: c_int {
+            #[cfg(target_os = "linux")]
+            SCHED_RESET_ON_FORK;
+        }
+    }
+
+    #[derive(Clone, Copy, Debug, Eq, Hash, PartialEq)]
+    pub struct SchedPolicy {
+        pub sched_type: SchedType,
+        #[cfg(target_os = "linux")]
+        pub sched_flags: SchedFlags,
+    }
+
+    impl SchedPolicy {
+        pub fn new(sched_type: SchedType) -> Self {
+            SchedPolicy::_with_flags(sched_type, SchedFlags::empty())
+        }
+
+        pub fn with_flags(sched_type: SchedType, sched_flags: SchedFlags) -> Self {
+            SchedPolicy::_with_flags(sched_type, sched_flags)
+        }
+
+        #[cfg(target_os = "linux")]
+        pub const fn bits(&self) -> i32 {
+            self.sched_type as i32 | self.sched_flags.bits()
+        }
+
+        #[cfg(not(target_os = "linux"))]
+        pub const fn bits(&self) -> i32 {
+            self.sched_type as i32
+        }
+
+        #[cfg(target_os = "linux")]
+        pub fn from_bits(bits: i32) -> Option<Self> {
+            let type_bits = bits & !SchedFlags::all().bits();
+            let sched_type = SchedType::try_from(type_bits).ok()?;
+            let flag_bits = bits & SchedFlags::all().bits();
+            let sched_flags = SchedFlags::from_bits(flag_bits)?;
+            Some(SchedPolicy::with_flags(sched_type, sched_flags))
+        }
+
+        #[cfg(not(target_os = "linux"))]
+        pub fn from_bits(bits: i32) -> Option<Self> {
+            let type_bits = bits & !SchedFlags::all().bits();
+            let sched_type = SchedType::try_from(type_bits).ok()?;
+            Some(SchedPolicy::new(sched_type))
+        }
+
+        #[cfg(target_os = "linux")]
+        #[inline]
+        fn _with_flags(sched_type: SchedType, sched_flags: SchedFlags) -> Self {
+            SchedPolicy {
+                sched_type,
+                sched_flags,
+            }
+        }
+
+        #[cfg(not(target_os = "linux"))]
+        #[inline]
+        fn _with_flags(sched_type: SchedType, _sched_flags: SchedFlags) -> Self {
+            SchedPolicy {
+                sched_type,
+            }
+        }
+    }
+
+    #[repr(transparent)]
+    #[derive(Clone, Copy, Debug, Eq, Hash, PartialEq)]
+    pub struct SchedParam(libc::sched_param);
+
+    impl SchedParam {
+        pub fn new(priority: i32) -> Self {
+            let mut sched_param: libc::sched_param = unsafe { mem::zeroed() };
+            sched_param.sched_priority = priority;
```

Comment:
```suggestion
            let sched_param = unsafe { libc::sched_param {
                sched_priority: priority,
                .. mem::zeroed()
            }};
```
```

Reply:
Hmm, looks like Illumos broke because of this. Illumos defines `libc::sched_param` as 

```rust
    pub struct sched_param {
        pub sched_priority: ::c_int,
        sched_pad: [::c_int; 8]
    }
```

Should I revert this change?

Reply:
Yeah.  Arguably, this is a bug in libc.  But I don't want to start that argument.  So instead you should do it like this: https://github.com/nix-rust/nix/blob/dab7332eabed8646f6d01a0d0688b4d1438accb4/src/sys/socket/mod.rs#L1388-L1401

Reply:
Think I have this working as expected now.